wrapper/src/main/java/software/amazon/jdbc/hostlistprovider/DynamicHostListProvider.java

```diff
@@ -16,7 +16,13 @@
 
 package software.amazon.jdbc.hostlistprovider;
 
-// A marker interface for providers that can fetch a host list reflecting the current database topology.
+import software.amazon.jdbc.HostSpec;
+
+// An interface for providers that can fetch a host list reflecting the current database topology.
 // Examples include providers for Aurora or Multi-AZ clusters, where the cluster topology, status, and instance roles
 // change over time.
-public interface DynamicHostListProvider extends HostListProvider { }
+public interface DynamicHostListProvider extends HostListProvider {
+  TopologyUtils getTopologyUtils();
```

Comment:
Does it make sense to replace these two methods with a single one? StaticHostListProvider can also implement this method.

`List<HostSpec> queryForTopology(conn, initialHostSpec)`